internal/cloudapi/openapi.yml

```diff
@@ -243,6 +253,36 @@ components:
           example: ['123456789012']
           items:
             type: string
+    GCPUploadRequestOptions:
+      type: object
+      required:
+        - bucket
+      properties:
+        region:
+          type: string
+          example: 'eu'
+          description: 'A valid GCP location. See https://cloud.google.com/storage/docs/locations'
+        bucket:
+          type: string
+          example: 'my-example-bucket'
+# don't expose the os type for now
```

Comment:
Any particular reason why/why not?

Reply:
Honestly, this is one thing that I wanted to discuss as part of the PR. The "os type" is used when importing an image into Compute Node from previously uploaded image from Storage. There are basically three options in the related behavior of the import:

1. You don't specify the "os type" and you also don't specify that the imported image should be treated as "data disk" - In this scenario the GCP import job will try to inspect the image and if it finds out that the OS is supported, it will modify it on the fly by installing some GCP repositories and packages. If the OS is not supported (e.g. Fedora), then the import fails.
2. You specify the "os type" - GCP will inspect the image, but will disregard the OS type and treat it as if it was the specified OS type. Modification and installation of repos and packages is handled the same way as in case (1.)
3. You specify that the image is "data disk" - GCP treats this image as if it was not bootable and does not try to modify it. This does not have any other effect on the usage of the image, other than that GCP won't be able to inject some information into it (like create user, inject SSH keys, ...) and won't be able to use RHUI infrastructure inside GCP.

Based on the discussion we had, we agreed that we don't want GCP to modify our images for now and that it is OK to expect that the image had user (with set SSH keys) created as part of the image build. So as a result, the code now imports images build through cloudapi always as "data disks".

My motivation to not expose the "os" in the cloudapi was to keep it minimal for now, so that we don't need to worry about backward compatibility and API stability if we decide to remove it or not use it. But other than that, it could be exposed if API stability is not really important for now...

Reply:
Makes sense! Let's discuss how to extend this after the first version is done.